framework/src/utils/MooseUtils.C

```diff
@@ -245,6 +245,12 @@ shortName(const std::string & name)
   return name.substr(name.find_last_of('/') != std::string::npos ? name.find_last_of('/') + 1 : 0);
 }
 
+std::string
+baseName(const std::string & name)
+{
+ return name.substr(0, name.find_last_of('/') != std::string::npos ? name.find_last_of('/') : 0);
+}
+
```

Comment:
Again, more and more parsing...

Let's just create a `TagObject` or something similar. You can append new tags in different formats and it handles all the storage and processing of the strings. Later you can pass that tag object another string and see if it matches or not.  We shouldn't see any parsing outside of this object. The MooseEnums work this way. You just initialize them and go. All of the string processing is handled internally.